src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -560,6 +560,10 @@ private void VisitSwitchExpressionCore(BoundSwitchExpression node, bool inferTyp
             var inferredState = BestTypeInferrer.GetNullableState(resultTypes);
             var resultType = TypeWithState.Create(inferredType, inferredState);
             inferredTypeWithAnnotations = resultType.ToTypeWithAnnotations();
+            if (resultType.State == NullableFlowState.MaybeDefault)
```

Comment:
> if (resultType.State == NullableFlowState.MaybeDefault) [](start = 11, length = 56)

Other scenarios where a type is inferred from a state would be useful to verify as well.
I see that you covered array creation `new[] { t, default}`. Maybe worth testing lambda and ternary (`b ? t : default`) as well, if not already covered?